operator/internal/handlers/internal/storage/secrets_test.go

```diff
@@ -169,7 +169,60 @@ func TestS3Extract(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "all set",
+			name: "unsupported SSE type",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint":          []byte("here"),
+					"bucketnames":       []byte("this,that"),
+					"access_key_id":     []byte("id"),
+					"access_key_secret": []byte("secret"),
+					"sse_type":          []byte("unsupported"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing SSE-KMS kms_key_id",
+			secret: &corev1.Secret{
+				Data: map[string][]byte{
+					"endpoint":                   []byte("here"),
+					"bucketnames":                []byte("this,that"),
+					"access_key_id":              []byte("id"),
+					"access_key_secret":          []byte("secret"),
+					"sse_type":                   []byte("SSE-KMS"),
+					"sse_kms_encryption_context": []byte("kms-encryption-ctx"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "all set with SSE-KMS",
```

Comment:
Do we need a `SSE-KMS without context` test (SSE-KMS without the one optional field)?